tests/Unit/Domain/CoordinateMaps/Test_ProductMapsTimeDep.cpp

```diff
@@ -0,0 +1,760 @@
+        {{functions_of_time.at(f_of_t_name)->func_and_deriv(time)[1][0], 0.0}});
+  }
+
+  {
+    using Map2d = CoordMapsTimeDependent::ProductOf2Maps<translation_map,
+                                                         translation_map>;
+
+    const std::string f_of_t_name_x{"translation_x"};
+    const std::string f_of_t_name_y{"translation_y"};
+
+    translation_map translation_map_x{f_of_t_name_x};
+    translation_map translation_map_y{f_of_t_name_y};
+    Map2d map2d(translation_map_x, translation_map_y);
+
+    functions_of_time[f_of_t_name_x] = std::make_unique<Polynomial>(
+        0.0,
+        std::array<DataVector, deriv_order + 1>{{{1.0}, {-3.0}, {2.0}, {0.0}}});
+    functions_of_time[f_of_t_name_y] = std::make_unique<Polynomial>(
+        0.0,
+        std::array<DataVector, deriv_order + 1>{{{1.0}, {2.4}, {2.0}, {0.0}}});
+
+    const double xA = -1.0;
+    const double xB = 1.0;
+    const double xa =
+        -1.0 + functions_of_time.at(f_of_t_name_x)->func(time)[0][0];
+    const double xb =
+        1.0 + +functions_of_time.at(f_of_t_name_x)->func(time)[0][0];
+
+    const double xi = 0.5;
+    const double x = xi + functions_of_time.at(f_of_t_name_x)->func(time)[0][0];
+    const double eta = 0.7;
+    const double y =
+        eta + functions_of_time.at(f_of_t_name_y)->func(time)[0][0];
+
+    const double yA = -1.0;
+    const double yB = 1.0;
+    const double ya =
+        -1.0 + functions_of_time.at(f_of_t_name_y)->func(time)[0][0];
+    const double yb =
+        1.0 + functions_of_time.at(f_of_t_name_y)->func(time)[0][0];
+
+    test_product_of_2_maps_time_dep(
+        map2d, time, functions_of_time, xA, xB, xa, xb, xi, x, eta, y, yA, yB,
+        ya, yb,
+        {{functions_of_time.at(f_of_t_name_x)->func_and_deriv(time)[1][0],
+          functions_of_time.at(f_of_t_name_y)->func_and_deriv(time)[1][0]}});
+    test_product_of_2_maps_time_dep(
+        serialize_and_deserialize(map2d), time, functions_of_time, xA, xB, xa,
+        xb, xi, x, eta, y, yA, yB, ya, yb,
+        {{functions_of_time.at(f_of_t_name_x)->func_and_deriv(time)[1][0],
+          functions_of_time.at(f_of_t_name_y)->func_and_deriv(time)[1][0]}});
+  }
+}
+
+template <typename Map1, typename Map2, typename Map3>
+void test_product_of_3_maps_time_dep(
+    const CoordMapsTimeDependent::ProductOf3Maps<Map1, Map2, Map3>& map3d,
+    const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time,
+    const double xA, const double xB, const double xa, const double xb,
+    const double xi, const double x, const double eta, const double y,
+    const double zeta, const double z, const double yA, const double yB,
+    const double ya, const double yb, const double zA, const double zB,
+    const double za, const double zb,
+    const std::array<double, 3>& expected_frame_velocity) noexcept {
+  const std::array<double, 3> point_A{{xA, yA, zA}};
```

Comment:
similar implementation to the product of 2 maps case, so any changes made for comments there should be made here too.